### PR TITLE
feat: add initial implementation for DIM wallet stub

### PR DESCRIPTION
**Pull request by nitin-vavdiya**
## Description

Initial contribution for SSI DIM wallet stub application
1. Spring boot application
2. README and other documentation
3. GitHub Workflows
4. GitHub action 
5. AUTHORS.md updated
6. Basic and working helm charts. No config added for Keycloak dependency

## Issue Link

Refs: [788](https://github.com/eclipse-tractusx/sig-release/issues/788)

## Checklist
Please delete options that are not relevant.

- [x] I have followed the contributing guidelines

- [ ] I have performed IP checks for added or updated 3rd party libraries

- [x] I have added copyright and license headers, footers (for .md files) or files (for images) //open source requirement

- [x] I have performed a self-review of my own code

- [x] I have successfully tested my changes locally

- [x] I have added tests and updated existing tests that prove my changes work

- [x] I have checked that new and existing tests pass locally with my changes

- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by nitin-vavdiya** [comment]
All secret alerts are false positives. These values are used as a sample request in API doc

**Comment by nitin-vavdiya** [comment]
IP ticket for wiremock lib: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15839

File: README.md [hunk 1]
```diff
@@ -1,5 +1,58 @@
-# ssi-dim-wallet-stub
+# SSI DIM Wallet stub application
 
-Stub application for Decentralize Identity Management (DIM) Wallet
+#### Note: *This application can not be used in production. This is designed and developed for local and integration testing only*
 
-### WIP
+The SSI Decentralize Identity Management (DIM)
```
Comment (evegufy) [inline README.md:5]: I think it should be called "Decentralized", the line break should be removed and might be good to explain the abbreviation SSI as well.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
+Wallet stub application is an alternative solution of wallet application to the Tractus-X.
```
Comment (evegufy) [inline README.md:6]: I don't understand that, there's no DIM wallet application from Tractus-X. The DIM is a non-FOSS solution by SAP.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,58 @@
-# ssi-dim-wallet-stub
+# SSI DIM Wallet stub application
 
-Stub application for Decentralize Identity Management (DIM) Wallet
+#### Note: *This application can not be used in production. This is designed and developed for local and integration testing only*
```
Comment (evegufy) [inline README.md:3]: As Tractus-X provides reference implementations only the reference to "production" is a bit misleading.
Maybe just leave it with "This application meant for **local setup** and - integration - **testing** only"

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+## Documentation for SSI DIM Wallet stub application
+
+#### Index
+
+1. How to contribute: [How to contribute.md](admin%2FDev%20Process%2FHow%20to%20contribute.md)
+2. Open API documentation: [openAPI.json](api%2FopenAPI.json)
+3. Architecture Documentation: [arc42](architecture%2Fmain.md)
+4. Integration with other components: [integration.md](admin%2Fintegration%2Fintegration.md)
```
Comment (evegufy) [inline docs/README.md:8]: I think this index is a bit weird as it does reflect the contents of /docs partially/very selectively

File: docs/admin/integration/integration.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+| BASE_WALLET_BPN | Base Wallet BPN | BPNL000000000000                       |
+| BASE_WALLET_DID | Base Wallet DID | did:web:localhost:BPNL000000000000     |
+| trustedIssuers  | Trusted Issuers | ["did:web:localhost:BPNL000000000000"] |
+
+## Notes
+
+- Ensure that all values are correctly set to the wallet stub application.
+- Update any missing or placeholder values with the actual configuration details.
+- For any additional configuration or troubleshooting, refer to the respective service documentation.
+
```
Comment (evegufy) [inline docs/admin/integration/integration.md:62]: Notice footer is missing

File: DOCKER_NOTICE.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+This application provides container images for demonstration purposes.
+
+DockerHub: To be added once we have Docker publish workflow
+
+Eclipse Tractus-X product(s) installed within the image:
+
+Item-Relationship-Service
+
+- GitHub: https://github.com/eclipse-tractusx/item-relationship-service
```
Comment (evegufy) [inline DOCKER_NOTICE.md:11]: the reference to IRS is wrong

File: DOCKER_NOTICE.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+- Project home: https://projects.eclipse.org/projects/automotive.tractusx
+- Dockerfile: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/blob/main/Dockerfile
+- Project
+  license: [Apache License, Version 2.0](https://github.com/eclipse-tractusx/item-relationship-service/blob/main/LICENSE)
```
Comment (evegufy) [inline DOCKER_NOTICE.md:15]: the reference to IRS is wrong

File: charts/ssi-dim-wallet-stub/Chart.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+description: |
+  A Helm chart to deploy SSI DIM wallet stub in kubernetes cluster
+type: application
+version: 0.1.0
+appVersion: "0.0.1"
+home: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/tree/main/charts/ssi-dim-wallet-stub
+sources:
+  - https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/tree/main/charts/ssi-dim-wallet-stub
+dependencies:
+  - name: keycloak
```
Comment (evegufy) [inline charts/ssi-dim-wallet-stub/Chart.yaml:31]: would it be possible to use the helm chart for centralidp directly https://github.com/eclipse-tractusx/portal-iam/tree/main/charts/centralidp?

File: charts/ssi-dim-wallet-stub/charts/keycloak-22.1.0.tgz
Comment (evegufy) [inline charts/ssi-dim-wallet-stub/charts/keycloak-22.1.0.tgz:1]: I wouldn't commit this, should be added to gitignore, see example https://github.com/eclipse-tractusx/portal-iam/blob/main/.gitignore#L3

File: charts/ssi-dim-wallet-stub/README.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
+## Helm char to deploy SSI DIM Wallet stub application
```
Comment (evegufy) [inline charts/ssi-dim-wallet-stub/README.md:1]: Typo in chart

File: .idea/codeStyles/codeStyleConfig.xml [hunk 10]
```diff
@@ -1,27 +1,5 @@
...
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-# License for the specific language governing permissions and limitations
-# under the License.
-#
-# SPDX-License-Identifier: Apache-2.0
-###############################################################
--->
-
```
Comment (evegufy) [inline .idea/codeStyles/codeStyleConfig.xml:21]: why did you remove the header again?

File: docs/architecture/main.md [hunk 11]
```diff
@@ -0,0 +1,1259 @@
+# **About arc42**
```
Comment (evegufy) [inline docs/architecture/main.md:1]: regarding the images under architecture/images: with reference to this TRG https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-4, would it be possible to move of the diagrams to mermaid or similar or to provide also some file which would allow editing in the future? as PNGs don't allow editing

File: README.md [hunk 12]
```diff
@@ -1,5 +1,58 @@
...
+3. If a wallet is not created at any point of request, application will create a new wallet at runtime
+4. Application will issue new VC during ``/presentations/query`` API if not issued previously
+5. Application will not give error if request VC is not already issue
+6. There is no separate IDP for this wallet application. OAuth token creation API is part of this application only
+7. We are not validating ``client_secret`` while creating OAuth token and  ``client_id`` will be BPN
+8. Negative scenarios are not covered
+9. ``jti`` claim is not validated
+10. No actual revocation of verifiable credentials
+11. All stored credentials will be lost on restart of the application
+12. JWTs are printed with debug log level for debugging purposes
```
Comment (evegufy) [inline README.md:58]: the notice for docker images should be linked, see example:https://github.com/eclipse-tractusx/portal-backend?tab=readme-ov-file#notice-for-docker-image

File: Dockerfile [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#################################################################################
+
```
Comment (evegufy) [inline Dockerfile:19]: The workflow for building the image is still missing right?
When adding the workflow could you please make sure to build multi platform? https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-08

One way would be using QEMU, like for instance PURIS does https://github.com/eclipse-tractusx/puris/blob/main/.github/workflows/build-image-backend.yml#L64

File: README.md [hunk 14]
```diff
@@ -1,5 +1,58 @@
...
 
-### WIP
+The SSI Decentralize Identity Management (DIM)
+Wallet stub application is an alternative solution of wallet application to the Tractus-X.
+
+This application can be used as a temporary wallet solution for local and integration testing.
+
+This application is using Java 21 and Spring boot framework to serve REST APIs.
+
+### Run in local
```
Comment (evegufy) [inline README.md:12]: Besides this, also the usage via helm chart should be mentioned or linked

File: charts/ssi-dim-wallet-stub/README.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+### Prerequisites
+- Kubernetes 1.19+
+- Helm 3.2.0+
+
+### Install
+```
+helm dep up charts/ssi-dim-wallet-stub
+kubectl create namespace wallet
+helm install wallet-stub -n wallet charts/ssi-dim-wallet-stub
+```
```
Comment (evegufy) [inline charts/ssi-dim-wallet-stub/README.md:22]: Please also mention the install of the released chart, see example https://github.com/eclipse-tractusx/puris/blob/main/charts/puris/README.md#install

**Review by evegufy — CHANGES_REQUESTED** [review]
Looks good in general, just some notes

File: README.md (hunk 1 above)
Comment (nitin-vavdiya) [reply README.md]: Addressed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/006e091ce7de7a4c265534389e3288e35968c732

File: README.md (hunk 2 above)
Comment (nitin-vavdiya) [reply README.md]: Modified readme: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/006e091ce7de7a4c265534389e3288e35968c732

File: README.md (hunk 3 above)
Comment (nitin-vavdiya) [reply README.md]: Addressed: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/006e091ce7de7a4c265534389e3288e35968c732

File: DOCKER_NOTICE.md (hunk 7 above)
Comment (nitin-vavdiya) [reply DOCKER_NOTICE.md]: Corrected: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/486556999761b417d2a1ee05e7615c6a1243080a

File: DOCKER_NOTICE.md (hunk 6 above)
Comment (nitin-vavdiya) [reply DOCKER_NOTICE.md]: Corrected: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/486556999761b417d2a1ee05e7615c6a1243080a

File: docs/admin/integration/integration.md (hunk 5 above)
Comment (nitin-vavdiya) [reply docs/admin/integration/integration.md:62]: Added with: Corrected: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/486556999761b417d2a1ee05e7615c6a1243080a

File: charts/ssi-dim-wallet-stub/charts/keycloak-22.1.0.tgz
Comment (nitin-vavdiya) [reply charts/ssi-dim-wallet-stub/charts/keycloak-22.1.0.tgz:1]: File removed and entry added in gitignore: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/9fec7a5c3c2463e00ee03823c75fda577114c3ac

File: .idea/codeStyles/codeStyleConfig.xml (hunk 10 above)
Comment (nitin-vavdiya) [reply .idea/codeStyles/codeStyleConfig.xml]: Added: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/69c58f27521831fc749979f9248d4ae836418a7e

File: charts/ssi-dim-wallet-stub/README.md (hunk 9 above)
Comment (nitin-vavdiya) [reply charts/ssi-dim-wallet-stub/README.md]: Fixed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/69c58f27521831fc749979f9248d4ae836418a7e

File: README.md (hunk 12 above)
Comment (nitin-vavdiya) [reply README.md:64]: Adressed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/3d0d3bb98b833e020fdb59e7a986a2e62b885018

File: README.md (hunk 14 above)
Comment (nitin-vavdiya) [reply README.md:18]: Added: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/80eaae7efb3b820dacfc844a1cbb35454076d014

File: Dockerfile (hunk 13 above)
Comment (nitin-vavdiya) [reply Dockerfile:19]: Yes, a workflow to build and publish docker image is not added yet.
Sure, will take a ref for a mentioned link.

File: charts/ssi-dim-wallet-stub/README.md (hunk 15 above)
Comment (nitin-vavdiya) [reply charts/ssi-dim-wallet-stub/README.md:29]: Added with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/64e4d7693197f3bb9a015787b9fbd46c47d7680c

File: docs/README.md (hunk 4 above)
Comment (nitin-vavdiya) [reply docs/README.md]: Modified with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/7ce8855d202a91dd166b799f989ac15740b98843

Let me know if further changes are necessary

File: charts/ssi-dim-wallet-stub/Chart.yaml (hunk 8 above)
Comment (nitin-vavdiya) [reply charts/ssi-dim-wallet-stub/Chart.yaml:31]: This application needs keycloak when it pushes did document to portal backend after a wallet is created, and API to push data in portal backend, we need to create access_token using from keycloak.

I avoided centralidp for further complexity. If it is necessary, then we can add it.

File: docs/architecture/main.md (hunk 11 above)
Comment (nitin-vavdiya) [reply docs/architecture/main.md:1]: There is one image system_scope.png that needs to be converted in diagram as code. we will check if this can be converted using mermaid. Is it okay if we push this change in the separate PR?

For add sequence diagram image plant UML is added
Other screenshots are added from swagger documentation.

File: src/main/resources/application.yaml [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+logging:
+  level:
+    org:
+      eclipse:
+        tractusx:
+          wallet:
+            stub: ${APP_LOG_LEVEL:DEBUG}
+stub:
+  env: ${STUB_ENV:local}
+  baseWalletBPN: ${BASE_WALLET_BPN:BPNL000000000000} #Oparator BPN
```
Comment (evegufy) [inline src/main/resources/application.yaml:68]: typo in Operator

File: charts/ssi-dim-wallet-stub/Chart.yaml (hunk 8 above)
Comment (evegufy) [reply charts/ssi-dim-wallet-stub/Chart.yaml:31]: I don't quite understand why adding centralidp Keycloak would add complexity, I think it would make things more easy to understand, for instance you even refer [here](https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/files#diff-8c14e17c4135ba3fa0819e6018b1fdba9edd22987bdec0e83f47fd25034f7c2fR130) to the CX-Central realm which is only available in centralidp Keycloak...
If you'd like to leave it with the generic Bitnami Keycloak, could you please add a comment here stating that the the equivalent Keycloak Instance in the Eclipse Tractus-X ecosystem is the centralidp Keycloak https://github.com/eclipse-tractusx/portal-iam/tree/v3.0.0/charts/centralidp

File: charts/ssi-dim-wallet-stub/README.md (hunk 15 above)
Comment (evegufy) [reply charts/ssi-dim-wallet-stub/README.md:29]: If you have added all information here, you could also remove the install.md on root level, or are you planning to fill it up with information?

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yaml (hunk 16 above)
Comment (nitin-vavdiya) [reply src/main/resources/application.yaml]: Fixed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/1042e142d1c22db74eeaf95c02b4b0c50f62f329

File: AUTHORS.md [hunk 17]
```diff
@@ -0,0 +1,10 @@
+# Authors
+
+The following people have contributed to this repository:
+
+* Julia Jeroch, Cofinity-X GmbH, https://github.com/jjeroch
+* Hakan Yildiz, Cofinity-X GmbH, https://github.com/hkny
+* Ronak Thacker, Cofinity-X GmbH, https://github.com/thackerronak
+* Leandro Willians Cavalcante Farias, Cofinity-X GmbH, https://github.com/leandro-cavalcante
+* Nitin Vavdiya, Cofinity-X GmbH, https://github.com/nitin-vavdiya
+
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply AUTHORS.md]: Removed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/1042e142d1c22db74eeaf95c02b4b0c50f62f329

File: README.md [hunk 18]
```diff
@@ -1,5 +1,77 @@
...
+12. JWTs are printed with debug log level for debugging purposes
+
+
+### Notice for Docker image
+
+This application provides container images for demonstration purposes.
+
+See Docker notice files for more information:
+
+- [SSI DIM Wallet stub Docker notice](DOCKER_NOTICE.md)
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply README.md:77]: License section added: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/1042e142d1c22db74eeaf95c02b4b0c50f62f329

File: README.md [hunk 19]
```diff
@@ -1,5 +1,77 @@
-# ssi-dim-wallet-stub
+# SSI DIM Wallet stub application
 
-Stub application for Decentralize Identity Management (DIM) Wallet
+#### Note: *This is meant and developed for local and integration testing only*
 
-### WIP
+The SSI Decentralized Identity Management (DIM) Wallet stub application provides REST APIs for below wallet functionality.
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply README.md]: Addressed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/1042e142d1c22db74eeaf95c02b4b0c50f62f329

File: README.md [hunk 20]
```diff
@@ -1,5 +1,77 @@
-# ssi-dim-wallet-stub
+# SSI DIM Wallet stub application
 
-Stub application for Decentralize Identity Management (DIM) Wallet
+#### Note: *This is meant and developed for local and integration testing only*
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply README.md]: Fixed: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/1042e142d1c22db74eeaf95c02b4b0c50f62f329

File: charts/ssi-dim-wallet-stub/README.md (hunk 15 above)
Comment (nitin-vavdiya) [reply charts/ssi-dim-wallet-stub/README.md:29]: INSTALL.md is removed: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/1042e142d1c22db74eeaf95c02b4b0c50f62f329

File: charts/ssi-dim-wallet-stub/Chart.yaml (hunk 8 above)
Comment (nitin-vavdiya) [reply charts/ssi-dim-wallet-stub/Chart.yaml]: centralidp added in dependencies : https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/42d0254d1400cc02f2974d41e5d6630c8e58ed0e

File: README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (evegufy) [reply README.md]: I don't see a sentence stating `The DIM Wallet is part of the Self-Sovereign Identity (SSI) Flow of Eclipse Tractus-X.`

File: README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (evegufy) [reply README.md:77]: For the README.md the license section is supposed to be like in the example example https://github.com/eclipse-tractusx/portal-backend?tab=readme-ov-file#license, please have look

File: README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply README.md]: I apologize for missing it.
Added with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/128e14d07fa9b4cfd70304befd66d3422d8ab0a2

File: README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply README.md:79]: Addressed with: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/889791d5c39385acd6685f39e760888f337241d4

File: README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (evegufy) [reply README.md]: @nitin-vavdiya no worries, looks good!

**Review by evegufy — CHANGES_REQUESTED** [review]
could you please make sure that your pr title aligns better with conventional commit messages? for instance `feat: add initial implementation for DIM wallet stub``
Also, please take care of the GitGuardian findings: you mentioned that they were false positives, with you committer rights you are enabled to mark them accordingly-

**Review by evegufy — DISMISSED** [review]
I also noticed that the following lib is still marked as restricted in the DEPENDENCIES file:

```
maven/mavencentral/org.wiremock/wiremock-standalone/3.9.1, (Apache-2.0 AND EPL-2.0) OR (Apache-2.0 AND GPL-2.0-only AND NOASSERTION), restricted, clearlydefined
```

Did you create an IP issue? If so, could you link it?

I think you should only open the IP issue for the project content / initial contribution once also the all 3rd party libs are vetted.

**Comment by evegufy** [comment]
@nitin-vavdiya I'm providing already some information for the review of the project content, which should only be started once you're sure that this PR remains as is / unchanged, so only once https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2#pullrequestreview-2233172056 is solved.

Relevant section in the Eclipse Foundation Handbook: https://www.eclipse.org/projects/handbook/#ip-project-content

As this PR represents the initial contribution and contains changes which were authored - not just by you (committer) but -also by contributors (see [Authors](https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/blob/889791d5c39385acd6685f39e760888f337241d4/AUTHORS.md)), it needs to undergo a review of Project Content at the  Eclipse Foundation, for which you (as committer) need to open an issue in Gitlab using the the according issue template, see https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/wikis/Requesting-a-Review.

Besides the `Basic Information`, please also provide additional information about why this review of Project Content is needed (like that this is an code initial contribution).

Here an example of a recent review of Project content, merely as example https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15836.
There it was also mentioned that it's a good idea to upload the state of the feature branch of the pull request (which isn't changed again) in a zip file to the IP issue in Gitlab, see https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15836#note_2617332, you can download a zip file by using the according GH feature:
image

**Comment by nitin-vavdiya** [comment]
> I also noticed that the following lib is still marked as restricted in the DEPENDENCIES file:
> 
> ```
> maven/mavencentral/org.wiremock/wiremock-standalone/3.9.1, (Apache-2.0 AND EPL-2.0) OR (Apache-2.0 AND GPL-2.0-only AND NOASSERTION), restricted, clearlydefined
> ```
> 
> Did you create an IP issue? If so, could you link it?
> 
> I think you should only open the IP issue for the project content / initial contribution once also the all 3rd party libs are vetted.

Thanks for pointing this out.

I already create IP check for the restricted lib: [IP issue](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15839)
Further, this is used for testing only, I refactor tests and removed this dependency. Ref: https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2/commits/8a041c3b4be3f2595da9e8aa42bf69e4e9108358

I closed this IP check issue for WireMock.

Now there is no any restricted lib in the code.

**Comment by nitin-vavdiya** [comment]
> could you please make sure that your pr title aligns better with conventional commit messages? for instance `feat: add initial implementation for DIM wallet stub` Also, please take care of the GitGuardian findings: you mentioned that they were false positives, with you committer rights you are enabled to mark them accordingly

PR title changed

I cannot see this repository in the GitGuardian app. maybe someone needs to give access.

**Comment by nitin-vavdiya** [comment]
> @nitin-vavdiya I'm providing already some information for the review of the project content, which should only be started once you're sure that this PR remains as is / unchanged, so only once [#2 (review)](https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/pull/2#pullrequestreview-2233172056) is solved.
> 
> Relevant section in the Eclipse Foundation Handbook: https://www.eclipse.org/projects/handbook/#ip-project-content
> 
> As this PR represents the initial contribution and contains changes which were authored - not just by you (committer) but -also by contributors (see [Authors](https://github.com/eclipse-tractusx/ssi-dim-wallet-stub/blob/889791d5c39385acd6685f39e760888f337241d4/AUTHORS.md)), it needs to undergo a review of Project Content at the Eclipse Foundation, for which you (as committer) need to open an issue in Gitlab using the the according issue template, see https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/wikis/Requesting-a-Review.
> 
> Besides the `Basic Information`, please also provide additional information about why this review of Project Content is needed (like that this is an code initial contribution).
> 
> Here an example of a recent review of Project content, merely as example https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15836. There it was also mentioned that it's a good idea to upload the state of the feature branch of the pull request (which isn't changed again) in a zip file to the IP issue in Gitlab, see https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15836#note_2617332, you can download a zip file by using the according GH feature: image

Thank you for the information.

IP check ticket is created: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15838

**Comment by nitin-vavdiya** [comment]
As a source code ZIP file is uploaded in [IP check](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15838), 
Additional changes will be made in separate pull requests, unless the changes are mandatory.

**Comment by evegufy** [comment]
> could you please make sure that your pr title aligns better with conventional commit messages? for instance `feat: add initial implementation for DIM wallet stub` Also, please take care of the GitGuardian findings: you mentioned that they were false positives, with you committer rights you are enabled to mark them accordingly

@nitin-vavdiya I'd find it odd if you couldn't manage the secret findings by GitGuardian with your committer rights.. don't you have the option to `Skip` when going to the Details of the check?

image
image

**Comment by nitin-vavdiya** [comment]
> > could you please make sure that your pr title aligns better with conventional commit messages? for instance `feat: add initial implementation for DIM wallet stub` Also, please take care of the GitGuardian findings: you mentioned that they were false positives, with you committer rights you are enabled to mark them accordingly
> 
> @nitin-vavdiya I'd find it odd if you couldn't manage the secret findings by GitGuardian with your committer rights.. don't you have the option to `Skip` when going to the Details of the check?
> 
> image image

Done, I was looking for resolve issue like button, so it will not rise the same issue again (Same as sonarQube maybe)
Thanks

**Comment by nitin-vavdiya** [comment]
Merging this PR based on [this comment](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15838#note_2818406)
